Code/Tools/AssetProcessor/native/ui/BuilderDataItem.cpp

```diff
@@ -0,0 +1,156 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project.
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+#include <native/ui/BuilderDataItem.h>
+
+namespace AssetProcessor
+{
+    const AZStd::string jobTypeDisplayNames[] = { "Analysis Jobs", "Processing Jobs" };
+    const AZStd::string invalidJobTypeDisplayName{ "Invalid Job Type" };
+
+    BuilderDataItem::BuilderDataItem(
+        ItemType itemType, const AZStd::string& name,
```

Comment:
you can pass AZStd::string by value and then use AZStd::move to move it into m_name. if you pass in a const char* on the other end to create a AZStd::string then it can steal the contents when its assigned to m_name.

https://stackoverflow.com/questions/51705967/advantages-of-pass-by-value-and-stdmove-over-pass-by-reference